### PR TITLE
feat(chip): add the chip component

### PR DESCRIPTION
**Pull request by adamraider**
image

### CODE REVIEW

File: packages/core/src/components/chip/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+class Chip {
```
Comment (adamraider) [inline packages/core/src/components/chip/index.js]: This needs tests before we merge.

File: packages/core/src/components/chip/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    this.state = {
+      active: options.active
+    };
+
+    this._bindEventListeners();
+    this.assignClasses();
+  }
+
+  _bindEventListeners() {
+    this._root.addEventListener('mousedown', this.onMousedown);
```
Comment (cmugla) [inline packages/core/src/components/chip/index.js]: Thoughts on adding a `touchstart` event as well.. not sure what the best UX is for this - seeing on my phone that they don't work very well for quicker, slightly clumsier gestures

File: packages/core/src/components/chip/index.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  assignClasses() {
+    if (this.state.active) {
+      this._root.classList.add('ray-chip--active');
+    } else {
+      this._root.classList.remove('ray-chip--active');
+    }
+  }
```
Comment (maniator) [inline packages/core/src/components/chip/index.js]: Can't this be done with plain css?

Comment (adamraider) [reply packages/core/src/components/chip/index.js]: I don't want to clutter the DOM with checkbox elements

**Comment by adamraider** [comment]
Closing in favor of #181

**Pull request closed without merging**